### PR TITLE
Helmet Storage Framework

### PR DESCRIPTION
**Pull request by Geevies**
* You can now store cigarette packets in helmets.

image
image
image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Review by SleepyGemmy — DISMISSED** [review]

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -25,10 +25,59 @@
...
+		return
+	return ..(over_object)
+
+/obj/item/clothing/head/helmet/handle_middle_mouse_click(mob/user)
+	if(has_storage && Adjacent(user))
+		hold.open(user)
+		return TRUE
+	return FALSE
+
+/obj/item/clothing/head/helmet/attackby(obj/item/W as obj, mob/user as mob)
```
Comment (NonQueueingMatt) [inline code/modules/clothing/head/helmet.dm:68]:
```suggestion
/obj/item/clothing/head/helmet/attackby(obj/item/W, mob/user)
```

File: code/modules/clothing/head/helmet.dm [hunk 2]
```diff
@@ -25,10 +25,59 @@
...
+			I.add_overlay(image('icons/clothing/kit/helmet_garb.dmi', null, thing_state))
+		I.add_overlay(image('icons/clothing/kit/helmet_garb.dmi', null, "helmet_band"))
+	return I
+
+/obj/item/clothing/head/helmet/attack_hand(mob/user as mob)
+	if(has_storage && !hold.handle_attack_hand(user))
+		return
+	return ..(user)
+
+/obj/item/clothing/head/helmet/MouseDrop(obj/over_object as obj)
```
Comment (NonQueueingMatt) [inline code/modules/clothing/head/helmet.dm:57]:
```suggestion
/obj/item/clothing/head/helmet/MouseDrop(obj/over_object)
```

File: code/modules/clothing/head/helmet.dm [hunk 3]
```diff
@@ -25,10 +25,59 @@
...
+	var/image/I = ..()
+	if(has_storage && slot == slot_head_str && length(hold.contents))
+		for(var/obj/item/thing as anything in hold.contents)
+			var/icon_type = hold.helmet_storage_types[thing.type]
+			var/thing_state = icon_type == HELMET_GARB_PASS_ICON ? initial(thing.icon_state) : icon_type
+			I.add_overlay(image('icons/clothing/kit/helmet_garb.dmi', null, thing_state))
+		I.add_overlay(image('icons/clothing/kit/helmet_garb.dmi', null, "helmet_band"))
+	return I
+
+/obj/item/clothing/head/helmet/attack_hand(mob/user as mob)
```
Comment (NonQueueingMatt) [inline code/modules/clothing/head/helmet.dm:52]:
```suggestion
/obj/item/clothing/head/helmet/attack_hand(mob/user)
```

File: code/modules/clothing/head/helmet.dm [hunk 4]
```diff
@@ -25,10 +25,59 @@
...
+	if(has_storage)
+		hold = new /obj/item/storage/internal/helmet(src)
+		hold.storage_slots = slots
+		hold.max_storage_space = slots * ITEMSIZE_SMALL
+		hold.max_w_class = ITEMSIZE_SMALL
+
+/obj/item/clothing/head/helmet/get_mob_overlay(var/mob/living/carbon/human/H, var/mob_icon, var/mob_state, var/slot)
+	var/image/I = ..()
+	if(has_storage && slot == slot_head_str && length(hold.contents))
+		for(var/obj/item/thing as anything in hold.contents)
```
Comment (NonQueueingMatt) [inline code/modules/clothing/head/helmet.dm:45]:
```suggestion
		for(var/obj/item/thing in hold.contents)
```

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Review by SleepyGemmy — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**